### ISSUE TITLE
SortedMap

### ISSUE THREAD

**Issue by kag0** [mrniko/redisson#309]
It would be useful to have a SortedMap implementation. For example in the case where entries might be accessed semi-regularly by a field which is not used as the key for the map. With a SortedMap the entry can be binary searched for in O(log(n)), but without you must either linear search in O(n) or maintain a separate map (which can be a pain to keep consistent).

Could this feature be added? or are there any nifty tricks with SortedSet to hack around the issue?

**Comment by mrniko** [comment]
Take a look at RedissonSortedSet it's all sync because of search of right index to insert implemented by Redisson not by redis. SortedMap, i think, would have same internal approach.

**Comment by kag0** [comment]
@mrniko makes sense. I suppose you could also build a O(log(n)) SortedMap implementation directly on top of RSortedSet, which might reduce some amount of code duplication since it would basically work the same internally.
